bundles/org.openhab.binding.emby/src/main/java/org/openhab/binding/emby/internal/protocol/EmbyClientSocket.java

```diff
@@ -0,0 +1,249 @@
+            if (json.has("Data")) {
+                JsonArray messageId = json.get("Data").getAsJsonArray();
+                Gson gson = new Gson();
+                Type type = new TypeToken<List<EmbyPlayStateModel>>() {
+                }.getType();
+                List<EmbyPlayStateModel> response = gson.fromJson(messageId, type);
+
+                response.forEach(playstate -> {
+
+                    // need to add internal validation method check but right now will pass all states to handler
+                    eventHandler.handleEvent(playstate);
+
+                });
+            }
+        }
+
+        @OnWebSocketClose
+        public void onClose(int statusCode, String reason) {
+            logger.debug("Closing a WebSocket due to {} with status code: {}", reason, Integer.toString(statusCode));
+            session = null;
+            connected = false;
+            if (eventHandler != null) {
+                scheduler.submit(() -> {
+                    try {
+                        eventHandler.onConnectionClosed();
+                    } catch (Exception e) {
+                        logger.debug("Error handling onConnectionClosed(): {}", e.getMessage(), e);
+                    }
+                });
+            }
+        }
+
+        @OnWebSocketError
+        public void onError(Throwable error) {
+            onClose(0, error.getMessage());
+        }
+
+    }
+
+    private void sendMessage(String str) throws IOException {
+        if (isConnected()) {
+            logger.debug("send message: {}", str);
+            session.getRemote().sendString(str);
+        } else {
+            throw new IOException("socket not initialized");
+        }
+    }
+
+    public @Nullable JsonElement callMethod(String methodName) {
+        return callMethod(methodName, null);
+    }
+
+    public @Nullable JsonElement callMethodString(String methodName, String dataParams) {
+        JsonObject payloadObject = new JsonObject();
+        payloadObject.addProperty("MessageType", methodName);
+        payloadObject.addProperty("Data", dataParams);
+        return callMethod(payloadObject);
+    }
+
+    public @Nullable JsonElement callMethod(String methodName, @Nullable JsonObject params) {
+        JsonObject payloadObject = new JsonObject();
+        // payloadObject.addProperty("jsonrpc", "2.0");
+        // payloadObject.addProperty("id", nextMessageId);
+        payloadObject.addProperty("MessageType", methodName);
+        if (params != null) {
+            payloadObject.add("Data", params);
+        }
+        return callMethod(payloadObject);
+    }
+
+    public synchronized @Nullable JsonElement callMethod(JsonObject payloadObject) {
+        try {
+            String message = mapper.toJson(payloadObject);
+            commandLatch = new CountDownLatch(1);
+            commandResponse = null;
+            sendMessage(message);
+            if (commandLatch.await(REQUEST_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
```

Comment:
This would block a thread in the frameworks shared thread pool. Is a timeout of 60 sec. necessary? Many bindings use 10 sec.

Reply:
i will change to 10 as that is the interval that the server sends the responses in anyways